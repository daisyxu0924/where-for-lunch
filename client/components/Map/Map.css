.root {
  width: 400px;

  & > div {
    height: 200px;
  }
}