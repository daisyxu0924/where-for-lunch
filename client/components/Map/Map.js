import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Map.css';

class Map extends PureComponent {
  static defaultProps = {
    zoom: 15, // Streets
  };

  static propTypes = {
    coordinates: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
    }),
    zoom: PropTypes.number,
  };

  state = {
    showMap: false,
  }

  componentDidMount() {
    if (this.props.coordinates && window.google) {
      const { latitude, longitude } = this.props.coordinates;
      const coordinatesObj = { lat: latitude, lng: longitude };

      this.map = new window.google.maps.Map(
        document.getElementById('map'),
        {
          center: coordinatesObj,
          zoom: 15,
        },
      );

      this.marker = new window.google.maps.Marker({
        position: coordinatesObj, map: this.map,
      });

      this.setState({ showMap: true });
    }
  }

  render() {
    return (
      <div className={styles.root}>
        <div id="map" hidden={!this.state.showMap}/>
      </div>
    );
  }
}

export default Map;
