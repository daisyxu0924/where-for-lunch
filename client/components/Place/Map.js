import React from 'react';
import PropTypes from 'prop-types';
import styles from './Place.css';

class Map extends React.PureComponent {
  componenDidMount() {
    this.initMap();
  }

  initMap() {
    console.log('google', google);
    google.maps.Map(document.getElementById('map'), { zoom: 8, center: this.props.coordinates });
  }

  render() {
    return (
      <div id="map" className={styles.map}></div>
    );
  }
}

Map.propTypes = {
  coordinates: PropTypes.object.isRequired,
};

export default Map;
