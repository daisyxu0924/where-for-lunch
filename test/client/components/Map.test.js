import React from 'react';
import { shallow } from 'enzyme';
import Map from 'client/components/Map/Map';
import renderer from 'react-test-renderer';

describe('Map Component', () => {
  const props = {
    coordinates: {
      latitude: 14.6823193,
      longitude: 121.0780226,
    },
  };

  test('renders hidden map when state.showMap = false', () => {
    const wrapper = shallow(
      <Map {...props} />,
    );

    wrapper.setState({ showMap: false });

    expect(wrapper.find('#map').prop('hidden')).toBe(true);
  });

  test('renders map when state.showMap = true', () => {
    const wrapper = shallow(
      <Map {...props} />,
    );

    wrapper.setState({ showMap: true });

    expect(wrapper.find('#map').prop('hidden')).toBe(false);
  });

  test('renders properly', () => {
    const mapComponent = renderer
      .create(<Map {...props} />)
      .toJSON();
    expect(mapComponent).toMatchSnapshot();
  });
});
