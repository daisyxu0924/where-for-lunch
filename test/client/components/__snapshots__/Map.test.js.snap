// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component renders properly 1`] = `
<div
  className="root"
>
  <div
    hidden={false}
    id="map"
  />
</div>
`;
